tensordict/nn/functional_modules.py

```diff
@@ -101,7 +101,7 @@ def _create_batched_inputs(flat_in_dims, flat_args, vmap_level: int, args_spec):
             arg
             if in_dim is None
             else arg.apply(
-                lambda _arg: _add_batch_dim(_arg, in_dim, vmap_level),  # noqa: B023
+                lambda _arg, in_dim=in_dim: _add_batch_dim(_arg, in_dim, vmap_level),
```

Comment:
Why does this work? Isn't `in_dim` still going to be a reference to the same mutable object?